tools/generator/generators.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+    "fmt"
+    "io"
+    "log"
+    "strings"
+    "encoding/binary"
+)
+
+type Vtx struct {
+  Ob [3]int16 /* x, y, z */
+  Flag  uint16
+  Tc [2]int16 /* texture coord */
+  Cn [4]uint8  /* colour & alpha */
+}
+
+func generate_model(textBuffer *strings.Builder, offset int, length int) {
+    textBuffer.WriteString("\nVtx some_var[] = {\n")
+
+    
+    _, err := rom.Seek(int64(offset), io.SeekStart)
+    if err != nil {
+        log.Fatal("Failed to seek during model generation: ", err)
+    }
+    
+    
+    for i := 0; i < length; i++ {
+        // Read the binary data into a Vtx struct
+        var vtx Vtx
+        err = binary.Read(rom, binary.BigEndian, &vtx)
+        if err != nil {
+            log.Fatal("Failed to read binary data:", err)
+        }
+        // Access the values in the Vtx struct
+        x := fmt.Sprint(vtx.Ob[0])
+        y := fmt.Sprint(vtx.Ob[1])
+        z := fmt.Sprint(vtx.Ob[2])
+        flag := fmt.Sprint(vtx.Flag)
+        tc1 := fmt.Sprint(vtx.Tc[0])
+        tc2 := fmt.Sprint(vtx.Tc[1])
+        cn1 := fmt.Sprintf("0x%02X", vtx.Cn[0])
+        cn2 := fmt.Sprintf("0x%02X", vtx.Cn[1])
+        cn3 := fmt.Sprintf("0x%02X", vtx.Cn[2])
+        cn4 := fmt.Sprintf("0x%02X", vtx.Cn[3])

```

Comment:
Probably more performant to write all of the Vtx into a struct var and then print that.